From 54738ad272e0631e776ebf4301a0a24bfd08b552 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Sat, 18 Aug 2018 14:39:31 +0530
Subject: [PATCH 0133/1051] soc: octeontx2: Setup resource limits before
 enabling VFs

Initialize resource limits before enabling AF's VFs.
Otherwise after enabling if VFs immediately request for resource
attach then it will fail.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/octeontx2/rvu.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/soc/marvell/octeontx2/rvu.c b/drivers/soc/marvell/octeontx2/rvu.c
index 8f703868228c..5f06b86307e7 100644
--- a/drivers/soc/marvell/octeontx2/rvu.c
+++ b/drivers/soc/marvell/octeontx2/rvu.c
@@ -2265,21 +2265,21 @@ static int rvu_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (err)
 		goto err_flr;
 
-	/* Enable AF's VFs (if any) */
-	err = rvu_enable_sriov(rvu);
+	err = rvu_policy_init(rvu);
 	if (err)
 		goto err_irq;
 
-	err = rvu_policy_init(rvu);
+	/* Enable AF's VFs (if any) */
+	err = rvu_enable_sriov(rvu);
 	if (err)
-		goto err_sriov;
+		goto err_policy;
 
 	/* Initialize debugfs */
 	rvu_dbg_init(rvu);
 
 	return 0;
-err_sriov:
-	rvu_disable_sriov(rvu);
+err_policy:
+	rvu_policy_destroy(rvu);
 err_irq:
 	rvu_unregister_interrupts(rvu);
 err_flr:
-- 
2.17.1

